[FRAME-180]: Refactor to remove "Mixed" status

tarecord opened the pull request:
This refactors the status to return a boolean whether the site is opted in or not. I also added some additional tests to verify things are working as expected.

<img width="745" alt="image" src="https://github.com/stellarwp/telemetry/assets/6947218/01a96ced-6f02-41ba-bf3f-79ac659f27d3">

ChrisMKindred commented on src/Telemetry/Opt_In/Status.php:
```diff
@@ -67,44 +64,38 @@ public function get_option() {
+	 * @since TBD   Update to remove unnecessary "mixed" status.
 	 *
-	 * @return integer The status value.
+	 * @return bool The site's current opt in status.
```
I think we need to keep this an integer in order to keep the same method signature and not cause a breaking change.

ChrisMKindred commented on src/Telemetry/Opt_In/Status.php:
```diff
@@ -67,44 +64,38 @@ public function get_option() {
+		 * @since TBD Update to return a bool status.
 		 *
-		 * @param integer $status The opt-in status value.
+		 * @param bool $status If the site is currently opted in.
```
Same as above, if we change this to a bool, it is a breaking change. Might be good to check with Stellar devs to make sure that isn't going to cause a problem.

ChrisMKindred reviewed (changes requested):
This this is a great cleanup for a confusing piece and the tests help make it very clear what is expected from the function.  I am concerned that we are changing the signature of a public method.  This is all good as long as Stellar is good with the breaking change that causes. Let's make sure we aren't causing a problem with that before merging.

tarecord commented:
@ChrisMKindred - Good point, I've reverted back to using integers (1 - Active, 2 - Inactive) and added another test to cover the `is_active()` method.

ChrisMKindred reviewed (approved):
👍  much better not requiring a depreciation now.
